src/sentry/tasks/merge.py

```diff
@@ -228,6 +229,50 @@ def rehash_group_events(group_id, transaction_id=None, **kwargs):
     delete_group.delay(group.id)
 
 
+def decrement_new_issue_count(group):
+    """
+    Updates the new issue/groups count
+
+    Should be called with the old group from merge_group.
+    That will be deleted, and so the count is decremented by 1.
+    """
+    from sentry.models import GroupRelease, Release, ReleaseProject, ReleaseProjectEnvironment
+    release_version = group.get_first_release()
+    if release_version is None:
+        return
+    try:
+        release = Release.objects.get(version=release_version)
+    except Release.DoesNotExist:
+        return
+    release.update(new_groups=release.new_groups - 1)
+
+    try:
+        release_project = ReleaseProject.objects.get(
+            release_id=release.id,
+            project_id=release.project_id,
+        )
+    except ReleaseProject.DoesNotExist:
+        # Done updating can't have a ReleaseProjectEnvironment without a ReleaseProject
+        return
+
+    release_project.update(new_groups=release_project.new_groups - 1)
+    release_project_envs = ReleaseProjectEnvironment.objects.filter(
+        release_id=release.id,
+        project_id=release.project_id,
+    ).select_related('environment')
+    group_releases_env_names = GroupRelease.objects.filter(
+        group_id=group.id,
+        release_id=release.id,
+        project_id=release.project_id,  # TODO(LB): not index on should this work?
+    ).values_list('environment', flat=True)
```

Comment:
I didn't test this so you'll want to double check but I think you can pass this to the `ReleaseProjectEnvironment` query above as `environment__name__in=GroupRelease.objects.filter(...)` to avoid having to make two separate queries and perform the join in the application.